include/evmc/evmc.h

```diff
@@ -140,55 +140,112 @@ typedef void (*evmc_get_block_hash_fn)(struct evmc_uint256be* result,
-    EVMC_BAD_JUMP_DESTINATION = 4,
-    EVMC_STACK_OVERFLOW = 5,
-    EVMC_STACK_UNDERFLOW = 6,
-    EVMC_REVERT = 7,                ///< Execution terminated with REVERT opcode.
-
-    /// Tried to execute an operation which is restricted in static mode.
-    EVMC_STATIC_MODE_VIOLATION = 8,
-
-    /// The dedicated INVALID instruction was hit.
-    EVMC_INVALID_INSTRUCTION = 9,
-
-    /// Tried to read outside memory bounds.
-    ///
-    /// An example is RETURNDATACOPY reading past the available buffer.
-    EVMC_INVALID_MEMORY_ACCESS = 10,
-
-    /// Exceptions produced by precompiles/system contracts
-    ///
-    /// An example: elliptic curve functions handed invalid EC points
-    EVMC_PRECOMPILE_FAILURE = 11,
-
-    /// Call depth exceded (if there is a call depth limit)
-    EVMC_CALL_DEPTH_EXCEDED = 12,
-
-    /// Contract validation has failed (e.g. due to EVM 1.5 jump validity,
-    /// Casper's purity checker or ewasm contract rules).
+/**
+ * The execution status code.
+ *
+ * Successful execution is represented by ::EVMC_SUCCESS having value 0.
+ *
+ * Positive values represent failures defined by VM specifications with generic
+ * ::EVMC_FAILURE code of value 1.
+ *
+ * Status codes with negative values represent VM internal errors
+ * not provided by EVM specifications. These errors MUST not be passed back
+ * to the caller. They MAY be handled by the Client in predefined manner
+ * (see e.g. ::EVMC_REJECTED), otherwise internal errors are not recoverable.
+ * The generic representant of errors is ::EVMC_INTERNAL_ERROR but
+ * an EVM implementation MAY return negative status codes that are not defined
+ * in the EVMC documentation.
+ *
+ * @note
+ * In case new status codes are needed, please create an issue or pull request
+ * in the EVMC repository (https://github.com/ethereum/evmc).
+ */
+enum evmc_status_code
+{
+    /** Execution finished with success. */
+    EVMC_SUCCESS = 0,
+
+    /** Generic execution failure. */
+    EVMC_FAILURE = 1,
+
+    /**
+     * Execution terminated with REVERT opcode.
+     *
+     * In this case the amount of gas left MAY be non-zero and additional output
+     * data MAY be provided in ::evmc_result.
+     */
+    EVMC_REVERT = 2,
+
+    /** The execution has run out of gas. */
+    EVMC_OUT_OF_GAS = 3,
+
+    /**
+     * The designated INVALID instruction has been hit during execution.
+     *
+     * The EIP-141 (https://github.com/ethereum/EIPs/blob/master/EIPS/eip-141.md)
+     * defines the instruction 0xfe as INVALID instruction to indicate execution
+     * abortion coming from high-level languages. This status code is reported
+     * in case this INVALID instruction has been encountered.
+     */
+    EVMC_INVALID_INSTRUCTION = 4,
+
+    /** An undefined instruction has been encountered. */
+    EVMC_UNDEFINED_INSTRUCTION = 5,
+
+    /**
+     * The execution has attempted to put more items on the EVM stack
+     * than the specified limit.
```

Comment:
Was the EVM stack limit 1024? If it is not going to change in the future, then it may be of use to include that number here 😁 @chfast

Reply:
Both limits for stack and call depth are 1024 and have never changed.

Reply:
It may make sense including in the description but not in the error name. We can consider it when a hardfork is introduced changing those :)

Reply:
I will leave it for another time...